pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -100,53 +125,50 @@ in
             ln -s ${withAllGrammars}/CONTRIBUTING.md .
             export ALLOWED_INSTALLATION_FAILURES=ipkg,norg,verilog
 
-            nvim --headless "+luafile ${withAllGrammars}/scripts/check-queries.lua" | tee log
+            nvim --headless -l "${withAllGrammars}/scripts/check-queries.lua" | tee log
 
             if grep -q Warning log; then
               echo "WARNING: warnings were emitted by the check"
               echo "Check if they were expected warnings!"
             fi
           '';
 
-      tree-sitter-queries-are-present-for-custom-grammars =
+      generated-queries-must-have-scm-files =
         let
-          pluginsToCheck =
-            map (grammar: grammarToPlugin grammar)
-              # non-derivations are here because there is a `recurseForDerivations = true`
-              (lib.filter lib.isDerivation (lib.attrValues tree-sitter-grammars));
+          queriesToCheck = builtins.attrValues queries;
         in
-        runCommand "nvim-treesitter-test-queries-are-present-for-custom-grammars" { CI = true; } ''
+        runCommand "nvim-treesitter-test-generated-queries-files" { CI = true; } ''
           touch "$out"
 
-          function check_grammar {
-            local grammar_name="$1"
-            local grammar_path="$2"
+          function check_query {
+            local lang="$1"
+            local path="$2"
+
+            echo "checking query files for $lang..."
+
+            # Check structure
+            if [ ! -d "$path/queries/$lang" ]; then
+               echo "Error: Directory queries/$lang does not exist in $path"
+               exit 1
+            fi
 
-            local grammar_queries=$(find -L "$grammar_path/queries/$grammar_name" -name '*.scm')
-            local EXPECTED_FILES="$grammar_path/parser/$grammar_name.so $grammar_queries"
-
-            echo ""
-            echo "expected files for $grammar_name:"
-            echo "$EXPECTED_FILES"
-
-            # the derivation has only symlinks, and `find` doesn't count them as files
-            # so we cannot use `-type f`, thus we use `-not -type d`
-            for file in $(find -L $2 -not -type d); do
-              echo "checking $file"
-              # see https://stackoverflow.com/a/8063284
-              if ! echo "$EXPECTED_FILES" | grep -wqF "$file"; then
-                echo "$file is unexpected, exiting"
-                exit 1
-              fi
-            done
+            # Check content
+            if [ -z "$(find -L "$path/queries/$lang" -name '*.scm')" ]; then
+               echo "Error: No .scm files found in $path/queries/$lang"
+               exit 1
+            fi
```

Comment:
question: Is this check necessary? `buildQueries` can create a queries derivation without any `.scm` files:

```bash
if [ -d "${super.nvim-treesitter.src}/runtime/queries/${language}" ]; then
  ln -s "${super.nvim-treesitter.src}/runtime/queries/${language}" "$out/queries/${language}"
else
  mkdir -p "$out/queries/${language}"
fi
```

Reply:
That is sort of the idea of the check, in case a query gets created without any content, the check will fail. The only way a query could be generated without is in case the nvim-treesitter source gets updated but the available queries doesn't, for example an language/filetype gets removed